Source/Core/Prune/Util.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Boogie
+{
+  public static class Util
+  {
+    public static V GetOrCreate<K, V>(this IDictionary<K, V> dictionary, K key, Func<V> createValue)
+    {
+      if (dictionary.TryGetValue(key, out var result)) {
+        return result;
+      }
+
+      result = createValue();
+      dictionary[key] = result;
+      return result;
+    }
+    
+    public static uint SafeMult(uint a, uint b) {
```

Comment:
The name "safe" here seems wrong to me. The method method is anything but safe, because it may not multiply at all. Instead, it returns the minimum of `MaxValue` and `a * b`. A name like `ApproximateMult` may be better, and adding a descriptive comment also seems good.

Reply:
Renamed it, not this was an existing method that I moved because it was in another class called Util.